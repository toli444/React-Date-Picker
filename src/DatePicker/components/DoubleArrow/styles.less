.double-arrow {
    border: 2px solid #2a2a2a;
    cursor: pointer;
    padding: 3px;
}
