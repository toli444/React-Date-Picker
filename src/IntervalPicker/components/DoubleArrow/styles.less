@hover-color: #dadada;

.double-arrow {
  cursor: pointer;
  padding: 3px;
  user-select: none;

  &:hover {
    background-color: #dadada;

    &:active {
      background-color: darken(@hover-color, 15%);
    }
  }
}
