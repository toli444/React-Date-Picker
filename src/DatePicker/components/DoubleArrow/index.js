import React from 'react';
import styles from './styles.css';

export default function DoubleArrow(props) {
  return <div className={styles['double-arrow']}>{props.left ? '<<' : '>>'}</div>;
};
