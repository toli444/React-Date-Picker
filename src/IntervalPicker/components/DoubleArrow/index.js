import React from 'react';
import styles from './styles.less';
import PropTypes from "prop-types";

const DoubleArrow = ({left, onClick}) => {
  return <div onClick={onClick} className={styles['double-arrow']}>{left ? '<<' : '>>'}</div>;
};

DoubleArrow.propTypes = {
  left: PropTypes.bool,
  onClick: PropTypes.func,
};

export default DoubleArrow;
