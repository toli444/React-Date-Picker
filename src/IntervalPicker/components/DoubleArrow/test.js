import React from 'react';
import DoubleArrow from '../DoubleArrow';
import { shallow } from 'enzyme';

describe('DoubleArrow', () => {
  it('renders correctly', () => {
    const wrapper  = shallow(<DoubleArrow />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders double left arrow if left', () => {
    const wrapper = shallow(<DoubleArrow left/>);

    expect(wrapper.get(0).props.children).toEqual('<<');
  });

  it('renders double right arrow otherwise', () => {
    const wrapper = shallow(<DoubleArrow />);

    expect(wrapper.get(0).props.children).toEqual('>>');
  });

  it ('calls correct function when click', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <DoubleArrow onClick={onClickMock}/>,
    );

    wrapper.simulate('click');

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
