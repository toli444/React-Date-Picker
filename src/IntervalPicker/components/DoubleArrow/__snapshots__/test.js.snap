// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoubleArrow renders correctly 1`] = `
<div
  className="double-arrow"
>
  &gt;&gt;
</div>
`;
